src/plays/smart-todo-list/App.js

```diff
@@ -0,0 +1,126 @@
+// All necessary dependences
+import { useEffect, useRef, useState } from 'react';
+import { BsPencil } from 'react-icons/bs';
+import Logo from './assets/logo.png';
+import Alert from './components/Alert';
+import DeleteModal from './components/DeleteModal';
+import Filter from './components/Filter';
+import TodoList from './components/TodoList';
+import { useGlobalContext } from './context/appContext';
+
+// App
+function App() {
+  // All necessary states
+  const [text, setText] = useState('');
+  const [isEditing, setIsEditing] = useState(false);
+  const [editID, setEditId] = useState(null);
+  const editRef = useRef();
+
+  const { isModalOpen, tasks, setTasks, filterdTasks, showAlert, alert, setAlert } =
+    useGlobalContext();
+
+  // Form submit handler
+  const submitHandler = (e) => {
+    // Reset default behavior
+    e.preventDefault();
+
+    if (!text) {
+      // if input is empty
+      showAlert(true, 'Field is required!', 'danger');
+    } else if (text && isEditing) {
+      // Update the item & states
+      const updatedTasks = tasks.map((item) => {
+        if (item.id === editID) {
+          return {
+            ...item,
+            title: text,
+            createdAt: new Date(),
+            complete: false
+          };
+        } else {
+          return item;
+        }
+      });
+      setTasks(updatedTasks);
+      setText('');
+      setEditId(null);
+      setIsEditing(false);
+
+      // Update item alert
+      setAlert({ show: true, msg: 'Updated Successfully!', type: 'success' });
+    } else {
+      // Add item
+      const newTask = {
+        id: tasks.length + 1,
+        createdAt: new Date(),
+        complete: false,
+        title: text
+      };
+      setTasks([newTask, ...tasks]);
+      setText('');
+
+      // Success Alert
+      showAlert(true, 'Added successfully!', 'success');
+    }
+  };
+
+  // Edit specific task
+  const editHandler = (id) => {
+    const spesificItem = tasks.find((item) => item.id === id);
+    setIsEditing(true);
+    setEditId(id);
+    setText(spesificItem.title);
+  };
+
+  useEffect(() => {
+    if (editRef.current) {
+      const end = editRef.current.value.length;
+      editRef.current.setSelectionRange(end, end);
+      editRef.current.focus();
+    }
+  }, [isEditing]);
+
+  return (
+    <section id="wa_todo-list">
+      {isModalOpen && <DeleteModal />}
+      <div className="wrapper">
+        <div className="wa_header">
```

Comment:
Check this background color again.. the log color.. somewhat has a submerge effect

![image](https://user-images.githubusercontent.com/3633137/210758739-47e20087-655d-4dc4-9dc9-e2c02ba6c573.png)

Reply:
Fixed